### DISCUSSION TITLE
Is there a way to add/remove/update dependencies for a specific workspace in a monorepo using pnpm?

### DISCUSSION THREAD

**Discussion by kitsune7** [pnpm/pnpm#4310] [Q&A] [answered]
Let's say I have workspace "a" and "b". If I want to install a new dependency like "lodash" to workspace a using npm, I could just run `npm install lodash -w a`, but in pnpm, I can't just swap out "npm install" for "pnpm add" and have it work the same way.

I know pnpm supports workspaces, but I'm not sure I understand how the `--workspace` option for `pnpm add` works. Judging from the description (and my attempts to work with it) however, it seems that the behavior is very different from the `--workspace` option in npm.

Am I missing something here, or do you have to run `cd $PATH_TO_WORKSPACE` before managing dependencies with pnpm? Is there a way to manage them from the root of the monorepo?

**Comment by zkochan** [comment, marked as answer]
`pnpm add lodash --filter a`

https://pnpm.io/filtering

**Reply by kitsune7** [reply]
That did it! Thanks @zkochan! I didn't realize that filtering could be used for workspaces!
